### PR TITLE
10914 new manager role

### LINKED ISSUE #10914: ✨ Define new Manager role in backend, and enable giving it to users.
## ✨ Feature

Add Manager role and associated permissions to rolepermission seeder. Ensure Platform Admins can give or take the role from any user.

## 🕵️ Details

As of this ticket, the Manager role doesn't actually do anything yet, so it doesn't need any permissions assigned to it beyond what base user already has. However, Platform Admins will need to be able to make other people managers, which means creating an `update-any-managerMembership` permission, giving it to platform_admins, and checking it appropriately where roles are assigned.

## 🙋‍♀️ Proposed Implementation

Add the new update-any-managerMembership permission in rolepermission.php and give it to platform admins. Update the UserPolicy updateRole method.

Update the Edit User admin ui page to ensure that Platform Admins can see if a user has a manager role, and give/take it from any user.

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [x] Manager role added to rolepermission seeder, but given no permissions yet.
- [x] Platform Admins given permission to assign Manager roles
- [x] UserPolicy updateRole check updated
- [x] Phpunit test to ensure platform admins, and no one else, can assign or remove Manager roles
- [x] On frontend, Platform Admins can assign or remove Manager role from users

## 🚀 Deployment

Run `php artisan db:seed --class=RolePermissionSeeder`

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #10914 

## 👋 Introduction

This branch adds a manager role that doesn't do anything yet.

## 🧪 Testing

1.  Rerun permission seeder: `php artisan db:seed --class=RolePermissionSeeder`

- [ ] AC on issue met
- [ ] Tests pass

It's tricky to manually check that another role cannot add or remove the manager permission since the pages (rightfully) won't load that would do it.  Here's the graphql query that can be run manually.

```graphql
mutation UpdateUserRoles($updateUserRolesInput: UpdateUserRolesInput!) {
  updateUserRoles(updateUserRolesInput: $updateUserRolesInput) {
    id
    roleAssignments {
      id
      role {
        id
        name
        isTeamBased
        displayName {
          en
          fr
          __typename
        }
        __typename
      }
      team {
        id
        name
        displayName {
          en
          fr
          __typename
        }
        __typename
      }
      __typename
    }
    __typename
  }
}
/****************/
{
  "updateUserRolesInput": {
    "roleAssignmentsInput": {
      "attach": [
        {
          "roleId": "0e9c0256-1002-48d4-9649-0cd2403c1cf3" // roles.id for manager role
        }
      ]
    },
    "userId": "1da1c55d-2708-48b1-8ec8-3658b7cbfda7" // users.id for user to change
  }
}
```

## 📸 Screenshot

image

## 🚚 Deployment

1.  Rerun permission seeder: `php artisan db:seed --class=RolePermissionSeeder`

### CODE REVIEW

File: api/tests/Unit/UserPolicyTest.php [hunk 1]
```diff
@@ -201,4 +202,43 @@ public function viewApplicant()
...
+            User::class,
+            [
+                'id' => $this->otherApplicant->id,
+                'roleAssignmentsInput' => [
+                    'attach' => [
+                        ['roleId' => $managerRoleId],
+                    ],
+                ],
+            ],
+        ];
```
Comment (petertgiles) [inline api/tests/Unit/UserPolicyTest.php:220]: This is pretty awkward.  I hope I'm doing it right.

**Review by tristan-orourke — COMMENTED** [review]
Looks fine to me

File: api/app/Policies/UserPolicy.php [hunk 2]
```diff
@@ -227,6 +227,9 @@ protected function individualAbleToCheck(User $actor, string $roleId)
                 return $actor->isAbleTo('assign-any-role');
             case 'platform_admin':
                 return $actor->isAbleTo('update-any-platformAdminMembership ') || $actor->isAbleTo('assign-any-role');
+            case 'manager':
+                return $actor->isAbleTo('update-any-managerMembership ') || $actor->isAbleTo('assign-any-role');
+
```
Comment (yonikid15) [inline api/app/Policies/UserPolicy.php:232]: Does this mean managers can update other manager memberships?

Comment (petertgiles) [reply api/app/Policies/UserPolicy.php:232]: It shouldn't!  :laughing:   It means that someone is able to add or remove the manager relationship if they have the update-any-managerMembership permission or the 'assign-any-role' permission.  That's configured in rolepermission.php

Comment (yonikid15) [reply api/app/Policies/UserPolicy.php:232]: Ahh okay now it makes sense :+1: :laughing:

**Review by yonikid15 — APPROVED** [review]
I tested everything and it looks great! Nice work!

**Pull request merged**